feat(clickhouse): Added max_suspicious_broken_parts to the config.xml

victorelec14 opened the pull request:
<!-- Describe your PR here. -->

This PR adds the "max_suspicious_broken_parts" setting which is necessary to imitate corrupted data in the Clickhouse database.

The default setting was 10, but it was changed to 100. In my case, setting it to 1000 solved the problems I had and allowed Clickhouse to start again normally.


Related Sentry Issue: #2832

Clickhouse References:
https://github.com/ClickHouse/ClickHouse/issues/41423
https://github.com/ClickHouse/ClickHouse/pull/41619
https://clickhouse.com/docs/en/operations/settings/merge-tree-settings

<!--

  Sentry employees and contractors can delete or ignore the following.

-->

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

azaslavsky commented:
How about we add the setting, but keep it at the default (10). Then above that, we can add a comment pointing to issue #2832 with a brief note about how you might adjust the setting to recover a corrupt install. That way future users will have a searchable way to fix this problem, but we won't mess up any active installs that may be relying on the old default.

victorelec14 commented:
> How about we add the setting, but keep it at the default (10). Then above that, we can add a comment pointing to issue #2832 with a brief note about how you might adjust the setting to recover a corrupt install. That way future users will have a searchable way to fix this problem, but we won't mess up any active installs that may be relying on the old default.

LGTM, I just updated the code and added a reference to the Issue, thanks !. 👍
